src/main/kotlin/tech/thatgravyboat/skyblockpv/screens/tabs/farming/ComposterScreen.kt

```diff
@@ -0,0 +1,293 @@
+import tech.thatgravyboat.skyblockpv.data.ComposterUpgrade
+import tech.thatgravyboat.skyblockpv.data.GardenProfile
+import tech.thatgravyboat.skyblockpv.data.StaticComposterData
+import tech.thatgravyboat.skyblockpv.data.StaticGardenData
+import tech.thatgravyboat.skyblockpv.utils.LayoutBuild
+import tech.thatgravyboat.skyblockpv.utils.Utils.append
+import tech.thatgravyboat.skyblockpv.utils.Utils.round
+import tech.thatgravyboat.skyblockpv.utils.components.PvWidgets
+import tech.thatgravyboat.skyblockpv.utils.displays.*
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+class ComposterScreen(gameProfile: GameProfile, profile: SkyBlockProfile? = null) : BaseFarmingScreen(gameProfile, profile) {
+    override fun getLayout() = LayoutBuild.horizontal {
+        widget(getPlots())
+        widget(
+            LayoutBuild.vertical {
+                widget(getInformation())
+                spacer(0, 10)
+                widget(getUpgrades())
+            },
+        )
+    }
+
+    private fun getInformation() = PvWidgets.label(
+        "Information",
+        LayoutBuild.vertical {
+            string("Organic Matter Stored: ") {
+                append(
+                    loadingComponent(
+                        gardenProfile,
+                        {
+                            Text.of(it.composterData.organicMatter.toLong().toFormattedString()) {
+                                this.color = TextColor.GREEN
+                            }
+                        },
+                    ),
+                )
+            }
+            string("Fuel Stored: ") {
+                append(
+                    loadingComponent(
+                        gardenProfile,
+                        {
+                            Text.of(it.composterData.fuel.toLong().toFormattedString()) {
+                                this.color = TextColor.DARK_GREEN
+                            }
+                        },
+                    ),
+                )
+            }
+            string("Compost Stored: ") {
+                append(
+                    loadingComponent(
+                        gardenProfile,
+                        {
+                            Text.of(it.composterData.compostItems.toFormattedString()) {
+                                this.color = TextColor.RED
+                            }
+                        },
+                    ),
+                )
+            }
+            string("Last Update: ") {
+                append(
+                    loadingComponent(
+                        gardenProfile,
+                        {
+                            val ofEpochMilli = Instant.ofEpochMilli(it.composterData.lastUpdateTimestamp)
+                            Text.of(
+                                DateTimeFormatter.ofPattern("uuuu/MM/dd HH:mm:ss").format(LocalDateTime.ofInstant(ofEpochMilli, ZoneOffset.systemDefault())),
```

Comment:
This should just be saved at the top instead of creating it every time

Reply:
have moved it to utils